docs/vmbackupmanager: add docs to clarify unsafe usage of lifecycle rules (#9728)
VictoriaMetrics/VictoriaMetrics: docs/victoriametrics/vmbackupmanager.md

* `-keepLastWeekly` - keep the last N weekly backups. Disabled by default
* `-keepLastMonthly` - keep the last N monthly backups. Disabled by default

> `0` value in every `keepLast*` flag results into deletion of ALL backups for particular type (hourly, daily, weekly and monthly)

> > Retention policy does not enforce removing previous versions of objects in object storages if versioning is enabled. See [these docs](https://docs.victoriametrics.com/victoriametrics/vmbackup/#permanent-deletion-of-objects-in-s3-compatible-storages) for more details.

Reviewer: ```suggestion
> Retention policy does not enforce removing previous versions of objects in object storages if versioning is enabled. See [permanent deletion of objects in s3 compatible-storages](https://docs.victoriametrics.com/victoriametrics/vmbackup/#permanent-deletion-of-objects-in-s3-compatible-storages) for more details.
```

> > It is possible enforce retention by using object storage lifecycle rules. Please, see [these docs](https://docs.victoriametrics.com/victoriametrics/vmbackupmanager/#retention-by-using-object-storage-lifecycle-rules) for more details.

Reviewer: ```suggestion
> It is possible enforce retention by using [object storage lifecycle rules](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html). Please, see [retention by using object storage lifecycle rules](https://docs.victoriametrics.com/victoriametrics/vmbackupmanager/#retention-by-using-object-storage-lifecycle-rules) for more details.
```

The result on the GCS bucket. We see only 3 daily backups:

![retention policy daily after retention cycle](vmbackupmanager_rp_daily_2.webp "retention policy daily after retention cycle")

#### Retention by using object storage lifecycle rules

> It is possible to enforce retention by using object storage lifecycle rules.

Reviewer: ```suggestion
It is possible to enforce retention by using [object storage lifecycle rules](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html).
```
